src/IssueViz.Security/OpenInIDE/Http/ShowHotspotRequestHandler.cs

```diff
@@ -0,0 +1,117 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Owin;
+using SonarLint.VisualStudio.Integration;
+using SonarLint.VisualStudio.IssueVisualization.Security.OpenInIDE.Contract;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.OpenInIDE.Http
+{
+    /// <summary>
+    /// Handles low-level HTTP request to open a hotspot
+    /// </summary>
+    [Export(typeof(IOwinPathRequestHandler))]
+    internal class ShowHotspotRequestHandler : IOwinPathRequestHandler
+    {
+        private const string ParamName_Server = "server";
+        private const string ParamName_Project = "project";
+        private const string ParamName_Hotspot = "hotspot";
+        private const string ParamName_Organization = "organization";
+
+        private readonly IOpenInIDERequestHandler openInIDERequestHandler;
+        private readonly ILogger logger;
+
+        [ImportingConstructor]
+        internal ShowHotspotRequestHandler(IOpenInIDERequestHandler openInIDERequestHandler, ILogger logger)
```

Comment:
The names are slightly confusing, both `ShowHotspotRequestHandler` and `IOpenInIDERequestHandler` are "request handlers", and it's not clear what distinction between them.

Perhaps `ShowHotspotRequestHandler` should be named `ShowHotspotHttpRequestHandler` or something of the sort.

Reply:
I'll change `ShowHotspotRequestHandler` to `ShowHotspotOwinRequestHandler`.

I don't think `IOpenInIDERequestHandler` should change to `IShowHotspotRequestHandler` since it's intended to handle all API calls, which we know will include handling vulnerabilities by the time of the LTS.

I was debating using "HTTP"/"Owin" in all of the zweb-aware components and "Api" in all of the web-agnostic components. I couldn't decide what to change `IOpenInIDERequestHandler` to, though. Maybe something like `ISonarLintApiHandler`.